applications/StructuralMechanicsApplication/tests/test_harmonic_analysis.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import print_function, absolute_import, division
+import KratosMultiphysics 
+
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+import KratosMultiphysics.ExternalSolversApplication as ExternalSolversApplication
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+from math import sqrt
+import os
+
+class ControlledExecutionScope:
+    def __init__(self, scope):
+        self.currentPath = os.getcwd()
+        self.scope = scope
+
+    def __enter__(self):
+        os.chdir(self.scope)
+
+    def __exit__(self, type, value, traceback):
+        os.chdir(self.currentPath)
+
+class HarmonicAnalysisTests(KratosUnittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def _add_variables(self,mp):
+        mp.AddNodalSolutionStepVariable(KratosMultiphysics.DISPLACEMENT)
+        mp.AddNodalSolutionStepVariable(KratosMultiphysics.REACTION)
+        mp.AddNodalSolutionStepVariable(KratosMultiphysics.TORQUE)
+        mp.AddNodalSolutionStepVariable(KratosMultiphysics.ROTATION)
+        mp.AddNodalSolutionStepVariable(KratosMultiphysics.VELOCITY)
+        mp.AddNodalSolutionStepVariable(KratosMultiphysics.ACCELERATION)
+        mp.AddNodalSolutionStepVariable(KratosMultiphysics.VOLUME_ACCELERATION)     
+        mp.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_MASS)   
+        mp.AddNodalSolutionStepVariable(StructuralMechanicsApplication.POINT_LOAD)
+
+    def _apply_material_properties(self,mp):
+        cl = StructuralMechanicsApplication.LinearElasticPlaneStress2DLaw()
+        mp.GetProperties()[1].SetValue(KratosMultiphysics.CONSTITUTIVE_LAW,cl)
+        mp.GetProperties()[1].SetValue(KratosMultiphysics.DENSITY,2000)
+        mp.GetProperties()[1].SetValue(KratosMultiphysics.YOUNG_MODULUS,10000)
+        mp.GetProperties()[1].SetValue(StructuralMechanicsApplication.CROSS_AREA,1.0)
+
+    def _solve_eigen(self,mp,echo=0):
+        eigensolver_settings = KratosMultiphysics.Parameters("""
+            {
+                    "solver_type": "FEAST",
+                    "print_feast_output": false,
+                    "perform_stochastic_estimate": false,
+                    "solve_eigenvalue_problem": true,
+                    "lambda_min": 1.0e-2,
+                    "lambda_max": 25.0,
+                    "search_dimension": 7,
+                    "linear_solver_settings": {
+                        "solver_type": "pastix"
```

Comment:
I would not use pastix, since it is not compiled by default
this makes the execution of the test unnecessarily difficult